### DISCUSSION TITLE
Unable to get cf_clearance cookie

### DISCUSSION THREAD

**Discussion by marcussacana** [cefsharp/CefSharp#4876] [General]
I'm trying to get all cookies from X instance to copy it to another RequestContext, after the user solve the cloudflare challange a cf_clearance is created with those headers:
```http
Set-Cookie: cf_clearance=; Path=/; Expires=Thu, 01-Jan-70 00:00:00 GMT; Domain=.domain.com; HttpOnly; Secure; SameSite=None
Set-Cookie: cf_clearance=USER_TOKEN; Path=/; Expires=Fri, 11-Jul-25 04:46:28 GMT; Domain=.domain.com; HttpOnly; Secure; SameSite=None; Partitioned
Set-Cookie: _va=13;path=/;domain=.domain.com;Max-Age=43200
```
Then I wrote this function to capture the cookies:
```csharp
public static CefSharp.Cookie[] GetCookies(this IBrowser Browser) {
    try
    {
        return AsyncContext.Run(async () =>
        {
            var Callback = new TaskCompletionCallback();
            var CookieManager = Cef.GetGlobalCookieManager(Callback);
            await Callback.Task;
            return await CookieManager.VisitAllCookiesAsync(); 
        }).ToArray();           
    }
    catch { return null;  }
}
```
But as you can see, there are no cf_clearance cookie
![image](https://github.com/cefsharp/CefSharp/assets/10576957/3e58135b-aecd-45e4-b080-a240330c066e)

Currently I was able to capture that cookie by using the `RequestHandler` and implementing a entirely manual http request for the browser, then sometime in the cloudflare startup the `cf_clearance` can be collected from the set-cookie header.
But this is not the best soluction because is too intrusive, my http request implementation isn't the same of the browser therefore the user has more trouble to solve the cloudflare challange.

**Comment by marcussacana** [comment]
I'm suspecting the problem is the cookie be set as `Secure` the CEF tries to hide it from the developer as well?

**Reply by amaitland** [reply]
Without a full example it's impossible to know what's going on. Remember that `CefSharp` is just a wrapper. It doesn't directly implement any of this functionality.

You can try use CDP to get the cookies.

- https://chromedevtools.github.io/devtools-protocol/tot/Storage/#method-getCookies
- https://github.com/cefsharp/CefSharp/issues/3165

**Comment by marcussacana** [comment]
A bit late, but I finally found the root cause of the problem.

What was happening is that my project was using .NET Framework, which only supports HTTP/1.1, and Cloudflare automatically suspects those requests are not coming from a real browser.

When I implemented a `RequestHandler` capture, I essentially wrote an HTTP client that replaced the browser's network communication. As a result, CEF was making HTTP/1.1 requests to Cloudflare instead of using its normal networking stack. This triggered Cloudflare's protection system and caused it to present the challenge.

When I disabled the `RequestHandler`, the browser used HTTP/2 instead, and Cloudflare no longer triggered the challenge. This also explains why the browser never had a `cf_clearance` cookie, because it wasn't needed in the first place.
